packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx

```diff
@@ -217,12 +311,47 @@ export function SelectNftTokenDialog(props: SelectNftTokenDialogProps) {
     } = props
     const { t } = useI18N()
     const account = useAccount()
-    const [tokenDetailed, setTokenDetailed] = useState<ERC721TokenDetailed>()
+    const [tokenDetailed, setTokenDetailed] = useState<ERC721TokenDetailed & { index: number }>()
     const [searched, setSearched] = useState(false)
     const [tokenDetailedSelectedList, setTokenDetailedSelectedList] =
-        useState<ERC721TokenDetailed[]>(existTokenDetailedList)
+        useState<(ERC721TokenDetailed & { index: number })[]>(existTokenDetailedList)
     const [loadingToken, setLoadingToken] = useState(false)
     const [tokenId, setTokenId, erc721TokenDetailedCallback] = useERC721TokenDetailedCallback(contract)
+    const [tokenIdListInput, setTokenIdListInput] = useState<string>('')
+    const [tokenIdFilterList, setTokenIdFilterList] = useState<string[]>([])
+    const [nonExistedTokenIdList, setNonExistedTokenIdList] = useState<string[]>([])
+    const [selectAll, setSelectAll] = useState<boolean>(false)
+    const maxSelectShares = Math.min(NFT_RED_PACKET_MAX_SHARES, tokenDetailedOwnerList.length)
+    const isSelectSharesExceed =
+        (tokenDetailedOwnerList.length === 0 ? NFT_RED_PACKET_MAX_SHARES - 1 : NFT_RED_PACKET_MAX_SHARES) <
+        tokenDetailedSelectedList.length
+    const { classes } = useStyles({ isSelectSharesExceed })
+    const selectAllHandler = useCallback(() => {
+        setTokenDetailedSelectedList(
+            tokenIdFilterList.length === 0
+                ? []
+                : selectAll
+                ? tokenDetailedSelectedList.filter((t) => !tokenIdFilterList.includes(t.tokenId))
+                : tokenDetailedSelectedList,
+        )
+
+        if (!selectAll) {
+            const newlyAdded = tokenDetailedOwnerList
+                .filter((t) => tokenIdFilterList.length === 0 || tokenIdFilterList.includes(t.tokenId))
+                .slice(0, maxSelectShares)
+            setTokenDetailedSelectedList(uniqBy(newlyAdded.concat(tokenDetailedSelectedList), 'tokenId'))
+        }
+
+        setSelectAll(!selectAll)
+    }, [selectAll, maxSelectShares, tokenIdFilterList, tokenDetailedSelectedList])
```

Comment:
`selectAllHandler` will be refreshed every time it executes, since `selectAll` is flipped inside it.

You could refactor `selectAll` as a computed variable instead of a state value:

```ts
const selectedAll = tokenDetailedSelectedList.length === allTokens.length
```